src/Workspaces/CSharp/Portable/FindSymbols/CSharpDeclaredSymbolInfoFactoryService.cs

```diff
@@ -186,153 +201,168 @@ public override bool TryGetDeclaredSymbolInfo(StringTable stringTable, SyntaxNod
+                        fullyQualifiedContainerName,
                         method.Modifiers.Any(SyntaxKind.PartialKeyword),
-                        IsExtensionMethod(method) ? DeclaredSymbolInfoKind.ExtensionMethod : DeclaredSymbolInfoKind.Method,
-                        GetAccessibility(method, method.Modifiers),
+                        isExtensionMethod ? DeclaredSymbolInfoKind.ExtensionMethod : DeclaredSymbolInfoKind.Method,
+                        GetAccessibility(container, method.Modifiers),
                         method.Identifier.Span,
                         inheritanceNames: ImmutableArray<string>.Empty,
                         parameterCount: method.ParameterList?.Parameters.Count ?? 0,
-                        typeParameterCount: method.TypeParameterList?.Parameters.Count ?? 0);
-                    return true;
+                        typeParameterCount: method.TypeParameterList?.Parameters.Count ?? 0));
+                    if (isExtensionMethod)
+                        AddExtensionMethodInfo(method, aliases, declaredSymbolInfos.Count - 1, extensionMethodInfo);
+                    return;
                 case SyntaxKind.PropertyDeclaration:
                     var property = (PropertyDeclarationSyntax)node;
-                    declaredSymbolInfo = DeclaredSymbolInfo.Create(
+                    declaredSymbolInfos.Add(DeclaredSymbolInfo.Create(
                         stringTable,
                         property.Identifier.ValueText, null,
-                        GetContainerDisplayName(node.Parent),
-                        GetFullyQualifiedContainerName(node.Parent),
+                        containerDisplayName,
+                        fullyQualifiedContainerName,
                         property.Modifiers.Any(SyntaxKind.PartialKeyword),
                         DeclaredSymbolInfoKind.Property,
-                        GetAccessibility(property, property.Modifiers),
+                        GetAccessibility(container, property.Modifiers),
                         property.Identifier.Span,
-                        inheritanceNames: ImmutableArray<string>.Empty);
-                    return true;
-                case SyntaxKind.VariableDeclarator:
-                    // could either be part of a field declaration or an event field declaration
-                    var variableDeclarator = (VariableDeclaratorSyntax)node;
-                    var variableDeclaration = variableDeclarator.Parent as VariableDeclarationSyntax;
-                    if (variableDeclaration?.Parent is BaseFieldDeclarationSyntax fieldDeclaration)
+                        inheritanceNames: ImmutableArray<string>.Empty));
+                    return;
+                case SyntaxKind.FieldDeclaration:
+                case SyntaxKind.EventFieldDeclaration:
+                    var fieldDeclaration = (BaseFieldDeclarationSyntax)node;
+                    foreach (var variableDeclarator in fieldDeclaration.Declaration.Variables)
```

Comment:
this logic got much simpler as we don't have to hit every vardecl in the file (which might be everywhere) and then filter to the subset that are fields/events.  instead we know if we've hit a field/event and can just process its vardecls directly.